mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -310,11 +470,25 @@ object StringIndexerModel extends MLReadable[StringIndexerModel] {
     override def load(path: String): StringIndexerModel = {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
-      val data = sparkSession.read.parquet(dataPath)
-        .select("labels")
-        .head()
-      val labels = data.getAs[Seq[String]](0).toArray
-      val model = new StringIndexerModel(metadata.uid, labels)
+
+      // We support to load old `StringIndexerModel` saved by previous Spark versions.
```

Comment:
nit, we can improve the phrasing a little bit here with "We support loading old..."

Reply:
I updated it.